homeassistant/components/bmp280/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for Bosch BMP280 Environmental Sensor integration."""
+import logging
+
+from adafruit_bmp280 import Adafruit_BMP280_I2C
+from busio import I2C
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.const import CONF_NAME, PRESSURE_HPA, TEMP_CELSIUS
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "BMP280"
+DEFAULT_I2C_ADDRESS = 0x77
+
+MIN_I2C_ADDRESS = 0x76
+MAX_I2C_ADDRESS = 0x77
+
+CONF_I2C_ADDRESS = "i2c_address"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): vol.All(
```

Comment:
Is this really optional. Should it be required?

Reply:
You're right, this is required as the hardware itself doesn't define a default I2C address